rpcserver.go

```diff
@@ -637,6 +637,70 @@ func createVinList(mtx *wire.MsgTx) []btcjson.Vin {
 	return vinList
 }
 
+// createVinList returns a slice of JSON objects for the inputs of the passed
+// transaction.
+func createVinListPrevOut(s *rpcServer, mtx *wire.MsgTx, chainParams *chaincfg.Params, vinExtra int) []btcjson.VinPrevOut {
+	vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
+
+	var txStore blockchain.TxStore
+	if vinExtra == 1 {
+		tx := btcutil.NewTx(mtx)
+		// Note: if all inputs are coinbase, then this extra DB lookup is
```

Comment:
I'd just do the coinbase check above all of this right after `vinList` and return early in that case.  There is no reason the check should be in the loop for every `TxIn`.  A coinbase transaction, by definition, only has a single `TxIn` entry.

I'd suggest:

```Go
// Coinbase transactions only have a single txin by definition.
vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
if blockchain.IsCoinBaseTx(mtx) {
	txIn := mtx.TxIn[0]
	vinList[0].Coinbase = hex.EncodeToString(txIn.SignatureScript)
	vinList[0].Sequence = txIn.Sequence
	return vinList
}

// Lookup all of the referenced transactions needed to populate the
// previous output information if requested.
var txStore blockchain.TxStore
if vinExtra != 0 {
	...
}

for i, txIn := range mtx.TxIn {
	vinList[i].Txid = txIn.PreviousOutPoint.Hash.String()
	vinList[i].Vout = txIn.PreviousOutPoint.Index
	vinList[i].Sequence = txIn.Sequence 

	...

	// Populate previous output information if requested.
	if vinExtra != 0 && len(txStore) > 0 {
		txData := txStore[v.PreviousOutPoint.Hash]
		...
	}
}
return vinList
```